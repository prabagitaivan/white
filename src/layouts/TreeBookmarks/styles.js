import { makeStyles } from '@material-ui/styles'

const layout = 'TreeBookmarks'

export const loading = makeStyles(
  {
    root: {
      display: 'flex',
      flexDirection: 'column',
      alignItems: 'center',
      marginTop: '40vh'
    },
    linearProgress: {
      height: 10,
      width: '60%',

      '& .MuiLinearProgress-barColorSecondary': {
        backgroundColor: '#b1b1b1'
      }
    }
  },
  { name: `${layout}Loading` }
)

export const empty = makeStyles(
  {
    root: {
      textAlign: 'center',
      marginTop: '40vh'
    }
  },
  { name: `${layout}Empty` }
)

export const tree = makeStyles(
  {
    root: {
      marginTop: -7.5,
      paddingTop: -7.5,
      paddingBottom: 10
    },
    tree: {
      '& .MuiTreeItem-group': {
        marginLeft: 7,
        paddingLeft: 14,
        borderLeft: `1px dashed gray`
      },
      '& .MuiTreeItem-label': {
        width: ({ desktop }) => (desktop ? '98%' : '91%')
      }
    },
    fontRoot: {
      fontSize: 20,
      marginTop: -1.5,
      fontWeight: 'bolder'
    },
    fontSubdata: {
      fontSize: 18,
      marginTop: -2.5
    },
    fontText: {
      fontSize: 15
    },
    fontUrl: {
      fontSize: 10,
      fontWeight: 'lighter'
    }
  },
  { name: `${layout}Tree` }
)
