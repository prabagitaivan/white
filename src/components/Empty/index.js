import React, { memo } from 'react'
import { makeStyles } from '@material-ui/styles'

const useStyles = makeStyles(
  {
    root: {
      textAlign: 'center',
      marginTop: '40vh'
    }
  },
  { name: 'Empty' }
)

export default memo(() => {
  const classes = useStyles()

  return (
    <div className={classes.root}>
      <div>Still Empty</div>
      <span role='img' aria-label='three-wise-monkeys'>
        ğŸ™ˆğŸ™‰ğŸ™Š
      </span>
    </div>
  )
})
