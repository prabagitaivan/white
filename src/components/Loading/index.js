import React, { memo } from 'react'
import { LinearProgress } from '@material-ui/core'
import { makeStyles } from '@material-ui/styles'

const useStyles = makeStyles(
  {
    root: {
      display: 'flex',
      flexDirection: 'column',
      alignItems: 'center',
      marginTop: '40vh'
    },
    linearProgress: {
      height: 10,
      width: '60%',

      '& .MuiLinearProgress-barColorSecondary': {
        backgroundColor: '#b1b1b1'
      }
    }
  },
  { name: 'Loading' }
)

export default memo(() => {
  const classes = useStyles()

  return (
    <div className={classes.root}>
      <div>Loading</div>
      <LinearProgress color='secondary' className={classes.linearProgress} />
    </div>
  )
})
