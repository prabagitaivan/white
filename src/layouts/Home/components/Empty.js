import React, { PureComponent } from 'react'
import { empty as styles } from '../styles'

export default class Empty extends PureComponent {
  render () {
    return (
      <div style={styles.root}>
        <div>Still Empty</div>
        <span role='img' aria-label='three-wise-monkeys'>
          ðŸ™ˆðŸ™‰ðŸ™Š
        </span>
      </div>
    )
  }
}
