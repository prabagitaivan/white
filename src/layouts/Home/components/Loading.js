import React, { PureComponent } from 'react'
import { LinearProgress } from '@material-ui/core'
import { loading as styles } from '../styles'

export default class Loading extends PureComponent {
  render () {
    return (
      <div style={styles.root}>
        <div>Loading</div>
        <LinearProgress color='secondary' style={styles.linearProgress} />
      </div>
    )
  }
}
