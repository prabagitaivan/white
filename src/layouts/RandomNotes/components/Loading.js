import React, { memo } from 'react'
import { LinearProgress } from '@material-ui/core'
import { loading as useStyles } from '../styles'

export default memo(() => {
  const classes = useStyles()

  return (
    <div className={classes.root}>
      <div>Loading</div>
      <LinearProgress color='secondary' className={classes.linearProgress} />
    </div>
  )
})
