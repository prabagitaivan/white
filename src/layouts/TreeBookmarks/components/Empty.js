import React, { memo } from 'react'
import { empty as useStyles } from '../styles'

export default memo(() => {
  const classes = useStyles()

  return (
    <div className={classes.root}>
      <div>Still Empty</div>
      <span role='img' aria-label='three-wise-monkeys'>
        ☁️🌧️⛈️
      </span>
    </div>
  )
})
