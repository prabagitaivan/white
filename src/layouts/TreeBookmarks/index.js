import React, { memo, useEffect } from 'react'
import { useSelector, useDispatch } from 'react-redux'
import {
  AddCircleOutline,
  BlurCircular,
  HighlightOff
} from '@material-ui/icons'
import { Grid } from '@material-ui/core'
import { TreeView, TreeItem } from '@material-ui/lab'
import { makeStyles } from '@material-ui/styles'
import Navigator from '../../components/Navigator'
import Content from '../../components/Content'
import { request } from '../../reducers/treeBookmarks'
import { setPage } from '../../reducers/status'
import Loading from '../../components/Loading'
import Empty from '../../components/Empty'
import Label from './components/Label'

const useStyles = makeStyles(
  {
    root: {
      marginTop: -7.5,
      paddingTop: -7.5,
      paddingBottom: 10
    },
    tree: {
      '& .MuiTreeItem-group': {
        marginLeft: 7,
        paddingLeft: 14,
        borderLeft: '1px dashed gray'
      },
      '& .MuiTreeItem-label': {
        width: ({ desktop }) => (desktop ? '98%' : '91%')
      }
    }
  },
  { name: 'TreeBookmarks' }
)

export default memo(() => {
  const { desktop } = useSelector(state => state.status)
  const { data, requesting } = useSelector(state => state.treeBookmarks)
  const dispatch = useDispatch()
  const classes = useStyles({ desktop })

  const openBookmark = url => window.open(url)

  useEffect(() => {
    dispatch(setPage('TreeBookmarks'))
    dispatch(request())
  }, [dispatch])

  return (
    <div>
      <Navigator />
      <Content>
        {requesting ? (
          <Loading />
        ) : Object.keys(data).length === 0 ? (
          <Empty />
        ) : (
          <Grid className={classes.root}>
            <TreeView
              className={classes.tree}
              defaultExpanded={['root']}
              defaultCollapseIcon={<HighlightOff />}
              defaultExpandIcon={<AddCircleOutline />}
              defaultEndIcon={<BlurCircular />}
            >
              <TreeItem nodeId='root' label={<Label type='root' />}>
                {Object.keys(data).map(subdata => (
                  <TreeItem
                    nodeId={subdata}
                    key={subdata}
                    label={<Label type='subdata' text={subdata} />}
                  >
                    {data[subdata]
                      .filter(bookmark => bookmark.active)
                      .map((bookmark, index) => (
                        <TreeItem
                          nodeId={`${subdata}-${index}`}
                          key={index}
                          label={
                            <Label text={bookmark.title} url={bookmark.url} />
                          }
                          onClick={() => openBookmark(bookmark.url)}
                        />
                      ))}
                  </TreeItem>
                ))}
              </TreeItem>
            </TreeView>
          </Grid>
        )}
      </Content>
    </div>
  )
})
