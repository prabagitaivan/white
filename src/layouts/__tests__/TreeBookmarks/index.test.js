import { render } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import React from 'react'
import { Provider } from 'react-redux'
import { createStore } from '../../../stores'
import Styles from '../../../styles'
import { request } from '../../../reducers/treeBookmarks'
import { setPage } from '../../../reducers/status'
import Loading from '../../../components/Loading'
import Empty from '../../../components/Empty'
import TreeBookmarks from '../../TreeBookmarks'

let store

function renderTreeBookmarks ({ desktop = true, treeBookmarks }) {
  const preloadedState = {
    status: { desktop, page: 'TreeBookmarks' },
    treeBookmarks
  }
  store = createStore(preloadedState)

  jest.spyOn(store, 'dispatch').mockReturnValue()

  return render(
    <Provider store={store}>
      <Styles>
        <TreeBookmarks />
      </Styles>
    </Provider>
  )
}

afterEach(() => {
  store.dispatch.mockRestore()
})

describe.only('layouts TreeBookmarks', () => {
  describe('snapshots', () => {
    it('contain navigator', () => {
      const { container } = renderTreeBookmarks({})
      const Navigator = container.querySelector('#navigator')
      expect(Navigator).toBeInTheDocument()
    })
    it('contain loading content when requesting', () => {
      const { container } = renderTreeBookmarks({})
      const LoadingRender = render(
        <Provider store={store}>
          <Styles>
            <Loading />
          </Styles>
        </Provider>
      )
      const Content = container.querySelector('#content')
      const LoadingContent = LoadingRender.container
      expect(Content.firstChild).toEqual(LoadingContent.firstChild)
    })
    it('contain empty content when requesting result is empty', () => {
      const { container } = renderTreeBookmarks({
        treeBookmarks: { requesting: false, data: {} }
      })
      const EmptyRender = render(
        <Provider store={store}>
          <Styles>
            <Empty />
          </Styles>
        </Provider>
      )
      const Content = container.querySelector('#content')
      const EmptyContent = EmptyRender.container
      expect(Content.firstChild).toEqual(EmptyContent.firstChild)
    })
    it('contain data content when requesting result is not empty in desktop', () => {
      const { container, queryByText } = renderTreeBookmarks({
        treeBookmarks: {
          requesting: false,
          data: {
            subdata1: [
              {
                title: 'title1',
                url: 'url1',
                active: true
              }
            ]
          }
        }
      })
      const Content = container.querySelector('#content')

      // root version
      expect(Content).toBeInTheDocument()
      expect(Content.firstChild).toMatchSnapshot()

      // subdata version
      const Data = queryByText('root')
      userEvent.click(Data)
      expect(Content.firstChild).toMatchSnapshot()

      // bookmark version
      const Subdata1 = queryByText('subdata1')
      userEvent.click(Subdata1)
      expect(Content.firstChild).toMatchSnapshot()
    })
    it('contain data content when requesting result is not empty in mobile', () => {
      const { container, queryByText } = renderTreeBookmarks({
        desktop: false,
        treeBookmarks: {
          requesting: false,
          data: {
            subdata1: [
              {
                title: 'title1',
                url: 'url1',
                active: true
              }
            ]
          }
        }
      })
      const Content = container.querySelector('#content')

      // root version
      expect(Content).toBeInTheDocument()
      expect(Content.firstChild).toMatchSnapshot()

      // subdata version
      const Data = queryByText('root')
      userEvent.click(Data)
      expect(Content.firstChild).toMatchSnapshot()

      // bookmark version
      const Subdata1 = queryByText('subdata1')
      userEvent.click(Subdata1)
      expect(Content.firstChild).toMatchSnapshot()
    })
    it('show active only in data content', () => {
      const { queryByText } = renderTreeBookmarks({
        treeBookmarks: {
          requesting: false,
          data: {
            subdata1: [
              {
                title: 'title1',
                url: 'url1',
                active: false
              },
              {
                title: 'title2',
                url: 'url2',
                active: true
              }
            ]
          }
        }
      })

      const Data = queryByText('root')
      userEvent.click(Data)

      const Subdata1 = queryByText('subdata1')
      userEvent.click(Subdata1)

      expect(queryByText('title1')).not.toBeInTheDocument()
      expect(queryByText('title2')).toBeInTheDocument()
    })
  })
  describe('mounting', () => {
    beforeEach(() => {
      renderTreeBookmarks({})
    })

    it('set page to tree bookmarks after mount', () => {
      expect(store.dispatch).toHaveBeenCalledWith(setPage('TreeBookmarks'))
    })
    it('request for tree bookmarks data after mount', () => {
      expect(store.dispatch).toHaveBeenCalledWith(request())
    })
  })
  describe('userEvent', () => {
    beforeEach(() => {
      jest.spyOn(window, 'open').mockReturnValue()
    })
    afterEach(() => {
      window.open.mockRestore()
    })

    it('open url when click the bookmark', () => {
      const { queryByText } = renderTreeBookmarks({
        treeBookmarks: {
          requesting: false,
          data: {
            subdata1: [
              {
                title: 'title1',
                url: 'url1',
                active: true
              },
              {
                title: 'title2',
                url: 'url2',
                active: true
              }
            ]
          }
        }
      })

      const Data = queryByText('root')
      userEvent.click(Data)

      const Subdata1 = queryByText('subdata1')
      userEvent.click(Subdata1)

      const Bookmark1 = queryByText('title1')
      userEvent.click(Bookmark1)
      expect(window.open).toHaveBeenCalledWith('url1')

      const Bookmark2 = queryByText('title2')
      userEvent.click(Bookmark2)
      expect(window.open).toHaveBeenCalledWith('url2')
    })
  })
})
