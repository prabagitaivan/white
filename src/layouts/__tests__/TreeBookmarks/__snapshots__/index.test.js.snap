// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`layouts TreeBookmarks snapshots contain data content when requesting result is not empty in desktop 1`] = `
<div
  class="MuiGrid-root WhiteTreeBookmarks-root"
>
  <ul
    aria-multiselectable="false"
    class="MuiTreeView-root WhiteTreeBookmarks-tree WhiteTreeBookmarks-tree"
    role="tree"
  >
    <li
      aria-expanded="false"
      class="MuiTreeItem-root"
      role="treeitem"
      tabindex="0"
    >
      <div
        class="MuiTreeItem-content"
      >
        <div
          class="MuiTreeItem-iconContainer"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
            />
          </svg>
        </div>
        <div
          class="MuiTypography-root MuiTreeItem-label MuiTypography-body1"
        >
          <p
            class="MuiTypography-root WhiteTreeBookmarksLabel-root MuiTypography-body1"
          >
            root
          </p>
        </div>
      </div>
    </li>
  </ul>
</div>
`;

exports[`layouts TreeBookmarks snapshots contain data content when requesting result is not empty in desktop 2`] = `
<div
  class="MuiGrid-root WhiteTreeBookmarks-root"
>
  <ul
    aria-multiselectable="false"
    class="MuiTreeView-root WhiteTreeBookmarks-tree WhiteTreeBookmarks-tree"
    role="tree"
  >
    <li
      aria-expanded="true"
      aria-selected="true"
      class="MuiTreeItem-root MuiTreeItem-expanded MuiTreeItem-selected"
      role="treeitem"
      tabindex="0"
    >
      <div
        class="MuiTreeItem-content"
      >
        <div
          class="MuiTreeItem-iconContainer"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M14.59 8L12 10.59 9.41 8 8 9.41 10.59 12 8 14.59 9.41 16 12 13.41 14.59 16 16 14.59 13.41 12 16 9.41 14.59 8zM12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
            />
          </svg>
        </div>
        <div
          class="MuiTypography-root MuiTreeItem-label MuiTypography-body1"
        >
          <p
            class="MuiTypography-root WhiteTreeBookmarksLabel-root MuiTypography-body1"
          >
            root
          </p>
        </div>
      </div>
      <ul
        class="MuiCollapse-container MuiTreeItem-group"
        role="group"
        style="min-height: 0px; height: 0px; transition-duration: 300ms;"
      >
        <div
          class="MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner"
          >
            <li
              aria-expanded="false"
              class="MuiTreeItem-root"
              role="treeitem"
              tabindex="-1"
            >
              <div
                class="MuiTreeItem-content"
              >
                <div
                  class="MuiTreeItem-iconContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiTypography-root MuiTreeItem-label MuiTypography-body1"
                >
                  <p
                    class="MuiTypography-root WhiteTreeBookmarksLabel-subdata MuiTypography-body1"
                  >
                    subdata1
                  </p>
                </div>
              </div>
            </li>
          </div>
        </div>
      </ul>
    </li>
  </ul>
</div>
`;

exports[`layouts TreeBookmarks snapshots contain data content when requesting result is not empty in desktop 3`] = `
<div
  class="MuiGrid-root WhiteTreeBookmarks-root"
>
  <ul
    aria-multiselectable="false"
    class="MuiTreeView-root WhiteTreeBookmarks-tree WhiteTreeBookmarks-tree"
    role="tree"
  >
    <li
      aria-expanded="true"
      class="MuiTreeItem-root MuiTreeItem-expanded"
      role="treeitem"
      tabindex="-1"
    >
      <div
        class="MuiTreeItem-content"
      >
        <div
          class="MuiTreeItem-iconContainer"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M14.59 8L12 10.59 9.41 8 8 9.41 10.59 12 8 14.59 9.41 16 12 13.41 14.59 16 16 14.59 13.41 12 16 9.41 14.59 8zM12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
            />
          </svg>
        </div>
        <div
          class="MuiTypography-root MuiTreeItem-label MuiTypography-body1"
        >
          <p
            class="MuiTypography-root WhiteTreeBookmarksLabel-root MuiTypography-body1"
          >
            root
          </p>
        </div>
      </div>
      <ul
        class="MuiCollapse-container MuiTreeItem-group"
        role="group"
        style="min-height: 0px; height: 0px; transition-duration: 300ms;"
      >
        <div
          class="MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner"
          >
            <li
              aria-expanded="true"
              aria-selected="true"
              class="MuiTreeItem-root MuiTreeItem-expanded MuiTreeItem-selected"
              role="treeitem"
              tabindex="0"
            >
              <div
                class="MuiTreeItem-content"
              >
                <div
                  class="MuiTreeItem-iconContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M14.59 8L12 10.59 9.41 8 8 9.41 10.59 12 8 14.59 9.41 16 12 13.41 14.59 16 16 14.59 13.41 12 16 9.41 14.59 8zM12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiTypography-root MuiTreeItem-label MuiTypography-body1"
                >
                  <p
                    class="MuiTypography-root WhiteTreeBookmarksLabel-subdata MuiTypography-body1"
                  >
                    subdata1
                  </p>
                </div>
              </div>
              <ul
                class="MuiCollapse-container MuiTreeItem-group"
                role="group"
                style="min-height: 0px; height: 0px; transition-duration: 300ms;"
              >
                <div
                  class="MuiCollapse-wrapper"
                >
                  <div
                    class="MuiCollapse-wrapperInner"
                  >
                    <li
                      class="MuiTreeItem-root"
                      role="treeitem"
                      tabindex="-1"
                    >
                      <div
                        class="MuiTreeItem-content"
                      >
                        <div
                          class="MuiTreeItem-iconContainer"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M10 9c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm0 4c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zM7 9.5c-.28 0-.5.22-.5.5s.22.5.5.5.5-.22.5-.5-.22-.5-.5-.5zm3 7c-.28 0-.5.22-.5.5s.22.5.5.5.5-.22.5-.5-.22-.5-.5-.5zm-3-3c-.28 0-.5.22-.5.5s.22.5.5.5.5-.22.5-.5-.22-.5-.5-.5zm3-6c.28 0 .5-.22.5-.5s-.22-.5-.5-.5-.5.22-.5.5.22.5.5.5zM14 9c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm0-1.5c.28 0 .5-.22.5-.5s-.22-.5-.5-.5-.5.22-.5.5.22.5.5.5zm3 6c-.28 0-.5.22-.5.5s.22.5.5.5.5-.22.5-.5-.22-.5-.5-.5zm0-4c-.28 0-.5.22-.5.5s.22.5.5.5.5-.22.5-.5-.22-.5-.5-.5zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm2-3.5c-.28 0-.5.22-.5.5s.22.5.5.5.5-.22.5-.5-.22-.5-.5-.5zm0-3.5c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1z"
                            />
                          </svg>
                        </div>
                        <div
                          class="MuiTypography-root MuiTreeItem-label MuiTypography-body1"
                        >
                          <p
                            class="MuiTypography-root WhiteTreeBookmarksLabel-title MuiTypography-body1 MuiTypography-noWrap"
                          >
                            title1
                          </p>
                          <p
                            class="MuiTypography-root WhiteTreeBookmarksLabel-url MuiTypography-body1 MuiTypography-noWrap"
                          >
                            url1
                          </p>
                        </div>
                      </div>
                    </li>
                  </div>
                </div>
              </ul>
            </li>
          </div>
        </div>
      </ul>
    </li>
  </ul>
</div>
`;

exports[`layouts TreeBookmarks snapshots contain data content when requesting result is not empty in mobile 1`] = `
<div
  class="MuiGrid-root WhiteTreeBookmarks-root"
>
  <ul
    aria-multiselectable="false"
    class="MuiTreeView-root WhiteTreeBookmarks-tree WhiteTreeBookmarks-tree"
    role="tree"
  >
    <li
      aria-expanded="false"
      class="MuiTreeItem-root"
      role="treeitem"
      tabindex="0"
    >
      <div
        class="MuiTreeItem-content"
      >
        <div
          class="MuiTreeItem-iconContainer"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
            />
          </svg>
        </div>
        <div
          class="MuiTypography-root MuiTreeItem-label MuiTypography-body1"
        >
          <p
            class="MuiTypography-root WhiteTreeBookmarksLabel-root MuiTypography-body1"
          >
            root
          </p>
        </div>
      </div>
    </li>
  </ul>
</div>
`;

exports[`layouts TreeBookmarks snapshots contain data content when requesting result is not empty in mobile 2`] = `
<div
  class="MuiGrid-root WhiteTreeBookmarks-root"
>
  <ul
    aria-multiselectable="false"
    class="MuiTreeView-root WhiteTreeBookmarks-tree WhiteTreeBookmarks-tree"
    role="tree"
  >
    <li
      aria-expanded="true"
      aria-selected="true"
      class="MuiTreeItem-root MuiTreeItem-expanded MuiTreeItem-selected"
      role="treeitem"
      tabindex="0"
    >
      <div
        class="MuiTreeItem-content"
      >
        <div
          class="MuiTreeItem-iconContainer"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M14.59 8L12 10.59 9.41 8 8 9.41 10.59 12 8 14.59 9.41 16 12 13.41 14.59 16 16 14.59 13.41 12 16 9.41 14.59 8zM12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
            />
          </svg>
        </div>
        <div
          class="MuiTypography-root MuiTreeItem-label MuiTypography-body1"
        >
          <p
            class="MuiTypography-root WhiteTreeBookmarksLabel-root MuiTypography-body1"
          >
            root
          </p>
        </div>
      </div>
      <ul
        class="MuiCollapse-container MuiTreeItem-group"
        role="group"
        style="min-height: 0px; height: 0px; transition-duration: 300ms;"
      >
        <div
          class="MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner"
          >
            <li
              aria-expanded="false"
              class="MuiTreeItem-root"
              role="treeitem"
              tabindex="-1"
            >
              <div
                class="MuiTreeItem-content"
              >
                <div
                  class="MuiTreeItem-iconContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M13 7h-2v4H7v2h4v4h2v-4h4v-2h-4V7zm-1-5C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiTypography-root MuiTreeItem-label MuiTypography-body1"
                >
                  <p
                    class="MuiTypography-root WhiteTreeBookmarksLabel-subdata MuiTypography-body1"
                  >
                    subdata1
                  </p>
                </div>
              </div>
            </li>
          </div>
        </div>
      </ul>
    </li>
  </ul>
</div>
`;

exports[`layouts TreeBookmarks snapshots contain data content when requesting result is not empty in mobile 3`] = `
<div
  class="MuiGrid-root WhiteTreeBookmarks-root"
>
  <ul
    aria-multiselectable="false"
    class="MuiTreeView-root WhiteTreeBookmarks-tree WhiteTreeBookmarks-tree"
    role="tree"
  >
    <li
      aria-expanded="true"
      class="MuiTreeItem-root MuiTreeItem-expanded"
      role="treeitem"
      tabindex="-1"
    >
      <div
        class="MuiTreeItem-content"
      >
        <div
          class="MuiTreeItem-iconContainer"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M14.59 8L12 10.59 9.41 8 8 9.41 10.59 12 8 14.59 9.41 16 12 13.41 14.59 16 16 14.59 13.41 12 16 9.41 14.59 8zM12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
            />
          </svg>
        </div>
        <div
          class="MuiTypography-root MuiTreeItem-label MuiTypography-body1"
        >
          <p
            class="MuiTypography-root WhiteTreeBookmarksLabel-root MuiTypography-body1"
          >
            root
          </p>
        </div>
      </div>
      <ul
        class="MuiCollapse-container MuiTreeItem-group"
        role="group"
        style="min-height: 0px; height: 0px; transition-duration: 300ms;"
      >
        <div
          class="MuiCollapse-wrapper"
        >
          <div
            class="MuiCollapse-wrapperInner"
          >
            <li
              aria-expanded="true"
              aria-selected="true"
              class="MuiTreeItem-root MuiTreeItem-expanded MuiTreeItem-selected"
              role="treeitem"
              tabindex="0"
            >
              <div
                class="MuiTreeItem-content"
              >
                <div
                  class="MuiTreeItem-iconContainer"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M14.59 8L12 10.59 9.41 8 8 9.41 10.59 12 8 14.59 9.41 16 12 13.41 14.59 16 16 14.59 13.41 12 16 9.41 14.59 8zM12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8z"
                    />
                  </svg>
                </div>
                <div
                  class="MuiTypography-root MuiTreeItem-label MuiTypography-body1"
                >
                  <p
                    class="MuiTypography-root WhiteTreeBookmarksLabel-subdata MuiTypography-body1"
                  >
                    subdata1
                  </p>
                </div>
              </div>
              <ul
                class="MuiCollapse-container MuiTreeItem-group"
                role="group"
                style="min-height: 0px; height: 0px; transition-duration: 300ms;"
              >
                <div
                  class="MuiCollapse-wrapper"
                >
                  <div
                    class="MuiCollapse-wrapperInner"
                  >
                    <li
                      class="MuiTreeItem-root"
                      role="treeitem"
                      tabindex="-1"
                    >
                      <div
                        class="MuiTreeItem-content"
                      >
                        <div
                          class="MuiTreeItem-iconContainer"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M10 9c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm0 4c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zM7 9.5c-.28 0-.5.22-.5.5s.22.5.5.5.5-.22.5-.5-.22-.5-.5-.5zm3 7c-.28 0-.5.22-.5.5s.22.5.5.5.5-.22.5-.5-.22-.5-.5-.5zm-3-3c-.28 0-.5.22-.5.5s.22.5.5.5.5-.22.5-.5-.22-.5-.5-.5zm3-6c.28 0 .5-.22.5-.5s-.22-.5-.5-.5-.5.22-.5.5.22.5.5.5zM14 9c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1zm0-1.5c.28 0 .5-.22.5-.5s-.22-.5-.5-.5-.5.22-.5.5.22.5.5.5zm3 6c-.28 0-.5.22-.5.5s.22.5.5.5.5-.22.5-.5-.22-.5-.5-.5zm0-4c-.28 0-.5.22-.5.5s.22.5.5.5.5-.22.5-.5-.22-.5-.5-.5zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8zm2-3.5c-.28 0-.5.22-.5.5s.22.5.5.5.5-.22.5-.5-.22-.5-.5-.5zm0-3.5c-.55 0-1 .45-1 1s.45 1 1 1 1-.45 1-1-.45-1-1-1z"
                            />
                          </svg>
                        </div>
                        <div
                          class="MuiTypography-root MuiTreeItem-label MuiTypography-body1"
                        >
                          <p
                            class="MuiTypography-root WhiteTreeBookmarksLabel-title MuiTypography-body1 MuiTypography-noWrap"
                          >
                            title1
                          </p>
                          <p
                            class="MuiTypography-root WhiteTreeBookmarksLabel-url MuiTypography-body1 MuiTypography-noWrap"
                          >
                            url1
                          </p>
                        </div>
                      </div>
                    </li>
                  </div>
                </div>
              </ul>
            </li>
          </div>
        </div>
      </ul>
    </li>
  </ul>
</div>
`;
